### PR TITLE
Introduce configurable delay

### LINKED ISSUE #30: vnet: Delay & Jitter options
## Summary
Introduce configurable delay & jitter applied when traversing each router.

## Motivation
To emulate more realistic network behavior.

## Proposal
Add following parameters to RouterConfig
* MinDelay (time.Duration)
* MaxJitter (time.Duration)

The actual delay would be:  minDelay + maxJitter * rand()  // where rand() returns [0, 1).

### PR DESCRIPTION
**Pull request by enobufs**
* Added following parameters to RouterConfig
  - MinDelay
  - MaxJitter
* Added a test for delay feature
* Resolves #30

### CODE REVIEW

File: vnet/router.go [hunk 1]
```diff
@@ -171,14 +180,25 @@ func (r *Router) Start() error {
...
-			select {
-			case <-r.pushCh:
-			case <-cancelCh:
-				break loop
+
+			if d <= 0 {
+				select {
+				case <-r.pushCh:
+				case <-cancelCh:
+					break loop
```
Comment (Sean-Der) [inline vnet/router.go]: What does the `break loop` do vs doing nothing at all?

Both times you will fall back into the `for` loop I believe

File: vnet/router_test.go [hunk 2]
```diff
@@ -242,6 +242,81 @@ func TestRouterStandalone(t *testing.T) {
 		assert.Equal(t, int32(1), atomic.LoadInt32(&nCbs1), "should be zero")
 		assert.Equal(t, byte(2), seq, "should be the last chunk")
 	})
+
+	t.Run("Delay with jitter", func(t *testing.T) {
```
Comment (Sean-Der) [inline vnet/router_test.go]: Mind doing smaller tests? Jitter, Delay and Jitter+Delay?

If you don't have the bandwidth I am happy to grab this! Just so people have a smaller test if they need to debug/fix in the future.

File: vnet/router.go (hunk 1 above)
Comment (enobufs) [reply vnet/router.go]: It's one of the confusing syntaxes in Go - and that's how you break out the for-loop, unlike to label for go-to..

* https://play.golang.org/p/HYqlhyNExD3
* https://medium.com/golangspec/labels-in-go-4ffd81932339

File: vnet/router_test.go (hunk 2 above)
Comment (enobufs) [reply vnet/router_test.go]: You mean, splitting the test I wrote into three?
```
	t.Run("Delay with jitter", func(t *testing.T)
```
I can do it!

Comment (enobufs) [reply vnet/router_test.go]: While attempting to write a test of "Jitter only", I noticed there's a case where the jitter value is not effective. Let me find out another way...

Comment (enobufs) [reply vnet/router_test.go]: @Sean-Der fixed. Let me know how you feel about the changes.

Comment (enobufs) [reply vnet/router_test.go]: By the way, I noticed golang's timer sleeps a bit longer than what is was told.  When timeout duration is set to 20ms, for instance, it wakes up in 20 ~ 24 ms... (a bit longer than I expected)

**Comment by enobufs** [comment]
@Sean-Der If you could review my changes per your feedback, that would be appreciated.

**Pull request merged**